### DISCUSSION TITLE
nvpmodel systemd service crashes

### DISCUSSION THREAD

**Discussion by jjannik** [OE4T/meta-tegra#1431] [Q&A] [answered]
Im running kirkstone with L4T35.3.1. Since I changed to a read-only filesystem the nvpmodel.service will fail (around 50%fail). Im wondering if this is critical or not. Thanks for any help. The gpu seeems to work in ether way (cuda-sample/devicequery)

Here the output after a failure

```
root@agx-orin:~# /usr/sbin/nvpmodel -f /etc/nvpmodel.conf --verbose
NVPM VERB: Config file: /etc/nvpmodel.conf
NVPM VERB: parsing done for /etc/nvpmodel.conf
NVPM VERB: set power mode as MODE_30W(2) in /etc/nvpmodel.conf.
NVPM ERROR: Error writing 240 to /sys/devices/gpu.0/tpc_pg_mask: 16
NVPM WARN: Reboot required for changing to this power mode: 2
NVPM WARN: DO YOU WANT TO REBOOT NOW? enter YES/yes to confirm:
```

```
root@agx-orin:~# cat /var/lib/nvpmodel/status
pmode:0002
```

Im aware of this Nvidia [link](https://docs.nvidia.com/jetson/archives/r35.3.1/DeveloperGuide/text/SD/PlatformPowerAndPerformance/JetsonOrinNanoSeriesJetsonOrinNxSeriesAndJetsonAgxOrinSeries.html#power-mode-controls) and the infos about `tpc_pg_mask` however Im not sure it this also true for meta-tegra.

**Comment by madisongh** [comment, marked as answer]
Yes, the same applies to Yocto builds - if you want to change the nvpmodel setting you do have to reboot, and you'll get that same message.

If nvpmodel.service is getting that error at boot time, then something else might be wrong, too.  There were some commits that went in back in July to fix some startup ordering issues that could cause spurious failures, so make sure you're up-to-date with at least commits 7479f93 and 5b426d3.

**Reply by jjannik** [reply]
Thank you so much! That seems to solve the issue. Sorry for the overhead.
